### ISSUE TITLE
Enigma failing on node v20.5.1

### ISSUE THREAD

**Issue by cognitivegears** [NuSkooler/enigma-bbs#490]
**Describe the Bug**
A clear and concise description of what the :bug: bug is.
enigma is failing to install or run with node v20.5.1 (and maybe others.) This is due to an old version of node-pty.

**To Reproduce**
Any relevant steps to reproduce the behavior:

nvm install v20.5.1
nvm use v20.5.1
/bin/rm -rf node_modules
npm install

**Expected Behavior**
modules install and enigma runs under v20.5.1

**Actual Behavior**
Fails to install modules or start (see below)

**Screenshots**

```
npm WARN skipping integrity check for git dependency ssh://git@github.com/NuSkooler/node-buffers.git
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm ERR! code 1
npm ERR! path /home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-pty
npm ERR! command failed
npm ERR! command sh -c node scripts/install.js
npm ERR! make: Entering directory '/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-pty/build'
npm ERR!   CXX(target) Release/obj.target/pty/src/unix/pty.o
npm ERR! make: Leaving directory '/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-pty/build'
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@8.4.1
npm ERR! gyp info using node@20.5.1 | linux | x64
npm ERR! gyp info find Python using Python version 3.10.12 found at "/usr/bin/python3"
npm ERR! gyp http GET https://nodejs.org/download/release/v20.5.1/node-v20.5.1-headers.tar.gz
npm ERR! gyp http 200 https://nodejs.org/download/release/v20.5.1/node-v20.5.1-headers.tar.gz
npm ERR! gyp http GET https://nodejs.org/download/release/v20.5.1/SHASUMS256.txt
npm ERR! gyp http 200 https://nodejs.org/download/release/v20.5.1/SHASUMS256.txt
npm ERR! gyp info spawn /usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-gyp/gyp/gyp_main.
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-pty/build/config.
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/cognitivegears/.cache/node-gyp/20.5.1/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/home/cognitivegears/.cache/node-gyp/20.5.1',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-gy
npm ERR! gyp info spawn args   '-Dnode_lib_file=/home/cognitivegears/.cache/node-gyp/20.5.1/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! In file included from ../../nan/nan.h:180,
npm ERR!                  from ../src/unix/pty.cc:20:
npm ERR! ../../nan/nan_callbacks.h:55:23: error: ‘AccessorSignature’ is not a member of ‘v8’
npm ERR!    55 | typedef v8::Local<v8::AccessorSignature> Sig;
npm ERR!       |                       ^~~~~~~~~~~~~~~~~
npm ERR! ../../nan/nan_callbacks.h:55:40: error: template argument 1 is invalid
npm ERR!    55 | typedef v8::Local<v8::AccessorSignature> Sig;
npm ERR!       |                                        ^
npm ERR! In file included from ../src/unix/pty.cc:20:
npm ERR! ../../nan/nan.h: In function ‘bool Nan::IdleNotification(int)’:
npm ERR! ../../nan/nan.h:686:63: warning: ‘bool v8::Isolate::IdleNotificationDeadline(double)’ is deprecated: Use MemoryPr
npm ERR!   686 |     return v8::Isolate::GetCurrent()->IdleNotificationDeadline(
npm ERR!       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
npm ERR!   687 |         idle_time_in_ms * 0.001);
npm ERR!       |         ~~~~~~~~~~~~~~~~~~~~~~~~
npm ERR! In file included from /home/cognitivegears/.cache/node-gyp/20.5.1/include/node/v8-initialization.h:13,
npm ERR!                  from /home/cognitivegears/.cache/node-gyp/20.5.1/include/node/v8.h:34,
npm ERR!                  from /home/cognitivegears/.cache/node-gyp/20.5.1/include/node/node.h:73,
npm ERR!                  from ../../nan/nan.h:60,
npm ERR!                  from ../src/unix/pty.cc:20:
npm ERR! /home/cognitivegears/.cache/node-gyp/20.5.1/include/node/v8-isolate.h:1293:8: note: declared here
npm ERR!  1293 |   bool IdleNotificationDeadline(double deadline_in_seconds);
npm ERR!       |        ^~~~~~~~~~~~~~~~~~~~~~~~
npm ERR! ../src/unix/pty.cc: In function ‘void pty_after_waitpid(uv_async_t*)’:
npm ERR! ../src/unix/pty.cc:512:9: warning: ‘void* memset(void*, int, size_t)’ writing to an object of type ‘class Nan::Persistent<v8::Function>’ with no trivial copy-assignment [-Wclass-memaccess]
npm ERR!   512 |   memset(&baton->cb, -1, sizeof(baton->cb));
npm ERR!       |   ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm ERR! In file included from ../../nan/nan.h:409,
npm ERR!                  from ../src/unix/pty.cc:20:
npm ERR! ../../nan/nan_persistent_12_inl.h:12:40: note: ‘class Nan::Persistent<v8::Function>’ declared here
npm ERR!    12 | template<typename T, typename M> class Persistent :
npm ERR!       |                                        ^~~~~~~~~~
npm ERR! In file included from ../../nan/nan.h:60,
npm ERR!                  from ../src/unix/pty.cc:20:
npm ERR! ../src/unix/pty.cc: At global scope:
npm ERR! /home/cognitivegears/.cache/node-gyp/20.5.1/include/node/node.h:1166:7: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
npm ERR!  1166 |       (node::addon_register_func) (regfunc),                          \
npm ERR!       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm ERR! /home/cognitivegears/.cache/node-gyp/20.5.1/include/node/node.h:1200:3: note: in expansion of macro ‘NODE_MODULE_X’
npm ERR!  1200 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
npm ERR!       |   ^~~~~~~~~~~~~
npm ERR! ../src/unix/pty.cc:734:1: note: in expansion of macro ‘NODE_MODULE’
npm ERR!   734 | NODE_MODULE(pty, init)g
npm ERR!       | ^~~~~~~~~~~
npm ERR! make: *** [pty.target.mk:113: Release/obj.target/pty/src/unix/pty.o] Error 1
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:294:12)
npm ERR! gyp ERR! System Linux 6.4.6-76060406-generic
npm ERR! gyp ERR! command "/home/cognitivegears/.nvm/versions/node/v20.5.1/bin/node" "/home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/.bin/node-gyp" "rebuild"
npm ERR! gyp ERR! cwd /home/cognitivegears/Documents/projects/bbs/enigma-bbs/node_modules/node-pty
npm ERR! gyp ERR! node -v v20.5.1
npm ERR! gyp ERR! node-gyp -v v8.4.1
npm ERR! gyp ERR! not ok
```

**Environment**
* [ x] I am using Node.js v14.x LTS or higher
* [ - ] `npm install` or `yarn` reports success
* Actual Node.js version (`node --version`):
* v20.5.1
* Operating system (`uname -a` on *nix systems):
Linux pop-os 6.4.6-76060406-generic #202307241739~1690928105~22.04~d567a38 SMP PREEMPT_DYNAM
IC Tue A x86_64 x86_64 x86_64 GNU/Linux

* Revision (`git rev-parse --short HEAD`):
b11ddfc3
* Any additional relevant information about your setup

Possible fix in PR #489

**Comment by cognitivegears** [comment]
This should be addressed in PR #489

**Comment by cognitivegears** [comment]
Closing as completed.
